### ISSUE TITLE
error compiling

### ISSUE THREAD

**Issue by samth** [pycket/pycket#280]
```
module-prefetch: (syntax/srcloc racket/pretty setup/path-to-relative racket/list) in: #<path:/home/samth/sw/plt/racket/collects/racket/contract/private/>        
  |  |  | processing: #<path:/home/samth/sw/plt/racket/collects/racket/pretty.rkt>
  |  |  | checking: #<path:/home/samth/sw/plt/racket/collects/racket/pretty.rkt>                                                                                 
  |  |  | old version 7.1.0.7 for #<path:/home/samth/sw/plt/racket/collects/racket/pretty.rkt>...                                                                
  |  |  | maybe-compile-zo starting #<path:/home/samth/sw/plt/racket/collects/racket/pretty.rkt>                                                                 
  |  |  |   no source hash: #<path:/home/samth/sw/plt/racket/collects/racket/pretty.rkt>                                                                         
  |  |  |   compiling #<path:/home/samth/sw/plt/racket/collects/racket/pretty.rkt>
module-prefetch: (racket/private/port racket/flonum racket/fixnum) in: #<path:/home/samth/sw/plt/racket/collects/racket/>                                        
  |  |  |   processing: #<path:/home/samth/sw/plt/racket/collects/racket/flonum.rkt>                                     
  |  |  |   checking: #<path:/home/samth/sw/plt/racket/collects/racket/flonum.rkt>                                                                              
  |  |  |   old version 7.1.0.7 for #<path:/home/samth/sw/plt/racket/collects/racket/flonum.rkt>...                                                              
  |  |  |   maybe-compile-zo starting #<path:/home/samth/sw/plt/racket/collects/racket/flonum.rkt>                                                               
  |  |  |  |  no source hash: #<path:/home/samth/sw/plt/racket/collects/racket/flonum.rkt>                                                                       
  |  |  |  |  compiling #<path:/home/samth/sw/plt/racket/collects/racket/flonum.rkt>                                                                             
module-prefetch: ((quote #%flfxnum) private/vector-wraps.rkt unsafe/ops.rkt) in: #<path:/home/samth/sw/plt/racket/collects/racket/>                              
  |  |  |  |  deleting #<path:/home/samth/sw/plt/racket/collects/racket/compiled/pycket/flonum_rkt.zo>
  |  |  |  |  failure                                                                                                                                            
exn:fail:contract : path-element->bytes expects path, got #<path:/>                                                                                              
  compilation context...:                                                                                                                                        
   #<path:/home/samth/sw/plt/racket/collects/racket/pretty.rkt>                                                                                                  
   #<path:/home/samth/sw/plt/racket/collects/racket/contract/private/blame.rkt>                                                                                 
   #<path:/home/samth/sw/plt/racket/collects/racket/contract/private/arrow-common.rkt>                                                                           
   #<path:/home/samth/sw/plt/racket/collects/racket/contract/base.rkt>
   #<path:/home/samth/sw/plt/racket/collects/compiler/cm.rkt>                                                                                                    
   #<path:/home/samth/sw/plt/pkgs/compiler-lib/compiler/commands/make.rkt>      
```

stack trace
```
-- ((lambda () (let ([main-path_1041 orig-path_1030]) (let ([alt-path_1042 (LinkletVar(rkt->ss:NO-VAL) orig-path_1030)][main-path-time_1043 (LinkletVar(try-file-time:NO-VAL) main-path_1041)]) (let ([alt-path-time_1044 (if (not main-path-time_1043) (if (not (eq? alt-path_1042 main-path_1041)) (LinkletVar(try-file-time:NO-VAL) alt-path_1042) #f) #f)]) (let ([path_1045 (if alt-path-time_1044 alt-path_1042 main-path_1041)][or-part_1047 main-path-time_1043]) (let ([path-time_1046 (if or-part_1047 or-part_1047 alt-path-time_1044)][path-zo-time_1048 (LinkletVar(get-compiled-time:NO-VAL) path->mode_1022 roots_1023 path_1045)][if291 (hash-ref seen_1028 path_1045 #f)]) (if if291 (begin (error 'compile-zo dependency cycle
  involves module: ~a path_1045) #f) (let ([if292 (if (not path-time_1046) (let ([AppRand9903 (current-compile-target-machine)]) (let ([or-part_1049 (not AppRand9903)]) (if or-part_1049 or-part_1049 (let ([AppRand9904 (read-deps_1031 path_1045)]) (LinkletVar(deps-machine:NO-VAL) Ap
-- ((lambda (trying-sha1?66_784 deps68_785 path->mode69_786 roots70_787 path71_788 orig-path72_789 read-src-syntax73_790 up-to-date74_791 collection-cache75_792
seen76_793) (let ([deps_794 deps68_785][path->mode_795 path->mode69_786][roots_796 roots70_787][path_797 path71_788][orig-path_798 orig-path72_789]) (let ([path_797 path_797][roots_796 roots_796][path->mode_795 path->mode_795][deps_794 deps_794][read-src-syntax_799 read-src-syntax73_790][up-to-date_800 up-to-date74_791][collection-cache_801 collection-cache75_792][seen_802 seen76_793][trying-sha1?_803 trying-sha1?66_784][actual-path_804 (LinkletVar(actual-source-path:NO-VAL) orig-path_798)]) (if trying-sha1?_803 (void) (let ([AppRator278 (LinkletVar(manager-compile-notify-handler:NO-VAL))]) (AppRator278 actual-path_804) (LinkletVar(trace-printf:NO-VAL) maybe-compile-zo starting ~a actual-path_804))) (begin0 (with-continuation-mark parameterization-key (let ([AppRand9834 (continuation-mark-set-first #f parameterization-key)][AppRan
-- ((lambda (assume-compiled-sha1238_830 just-touch?236_831 recompile-from237_832 use-existing-deps239_833) (let ([just-touch?_834 just-touch?236_831][recompile-from_835 recompile-from237_832][assume-compiled-sha1_836 assume-compiled-sha1238_830][use-existing-deps_837 use-existing-deps239_833][lc_838 (LinkletVar(parallel-lock-client:NO-VAL))]) (let ([ (begin (if lc_838 (LinkletVar(log-compile-event:NO-VAL) path_797 'locking) (void)) (values))][locked?_839 (if lc_838 (lc_838 'lock zo-name_805) #f)][or-part_841 (not lc_838)]) (let ([ok-to-compile?_840 (if or-part_841 or-part_841 locked?_839)]) (dynamic-wind (lambda () (void)) (lambda () (if ok-to-compile?_840 (let ([if254 (if CellRef(just-touch?_834) (file-exists? zo-name_805) #f)]) (if if254 (begin (LinkletVar(log-compile-event:NO-VAL) path_797
'start-touch) (LinkletVar(1/touch:NO-VAL) zo-name_805)) (begin (if CellRef(just-touch?_834) (set! CellRef(just-touch?_834) #f) (void)) (let ([AppRand9838 (if recompile-from_835 'start-recompile 'sta
-- ((lambda () (if ok-to-compile?_840 (let ([if254 (if CellRef(just-touch?_834) (file-exists? zo-name_805) #f)]) (if if254 (begin (LinkletVar(log-compile-event:NO-VAL) path_797 'start-touch) (LinkletVar(1/touch:NO-VAL) zo-name_805)) (begin (if CellRef(just-touch?_834) (set! CellRef(just-touch?_834) #f) (void)) (let ([AppRand9838 (if recompile-from_835 'start-recompile 'start-compile)]) (LinkletVar(log-compile-event:NO-VAL) path_797 AppRand9838)) (let ([AppRand9839 (if recompile-from_835 re )]) (LinkletVar(trace-printf:NO-VAL) ~acompiling ~a AppRand9839 actual-path_804)) (with-continuation-mark parameterization-key (let ([AppRand9840 (continuation-mark-set-first #f parameterization-key)][AppRand9841 (LinkletVar(depth:NO-VAL))]) (let ([AppRand9842 (+ AppRand9841 1)]) (extend-parameterization AppRand9840 LinkletVar(depth:NO-VAL) AppRand9842))) (let ([with-handlers-predicate247_842 LinkletVar(exn:get-module-code?:NO-VAL)][with-handlers-handler248_843 (lambda (ex_844) (let ([AppRand9843 (Linkl
```

**Comment by samth** [comment]
The issue is converting this path: `#<path:/home/samth/sw/plt/racket/collects/racket/private//../vector-wraps.rkt>` in the fasl code.

**Comment by samth** [comment]
```
> (explode-path (build-path "/home/samth/sw/plt/racket/collects/racket/private/" 'up  "vector-wraps.rkt"))
(#<path:/> #<path:home> #<path:samth> #<path:sw> #<path:plt> #<path:racket> #<path:collects> #<path:racket> #<path:private> #<path:/> up #<path:vector-wraps.rkt>)
>
[samth@huor:~/.../racket/collects/racket (master) plt] r
Welcome to Racket v7.1.0.10.
> (explode-path (build-path "/home/samth/sw/plt/racket/collects/racket/private/" 'up  "vector-wraps.rkt"))
'(#<path:/>
  #<path:home>
  #<path:samth>
  #<path:sw>
  #<path:plt>
  #<path:racket>
  #<path:collects>
  #<path:racket>
  #<path:private>
  up
  #<path:vector-wraps.rkt>)
```

**Comment by samth** [comment]
Fixed by d45d72e9b164a2ad6f
